common.mk

```diff
@@ -1,33 +1,81 @@
 # Common makefile shared between Teleport OSS and Ent.
 
+# -----------------------------------------------------------------------------
+# pkg-config / pkgconf
+# Set $(PKGCONF) to either pkgconf or pkg-config if either are installed
+# or /usr/bin/false if not. When it is set to "false", running $(PKGCONF)
+# will exit non-zero with no output.
+
+PKGCONF := $(firstword $(shell which pkgconf pkg-config false))
+
+# -----------------------------------------------------------------------------
+# libbpf detection
+#
+# Requirements for building with BFP support:
+# * clang and llvm-strip programs
+# * linux on amd64 or amd64
+# * libbpf 1.2.2 (either in /usr/libbpf-1.2.2 or pkg-config)
+#   * The centos7 buildbox puts this in /usr/libbpf-1.2.2
+#   * The ng buildbox has a pkg-config file for it
+#   * Native/local builds have a pkg-config file for it
+# * Either using a cross-compiling buildbox or is a native build
+#
+# The default is without BPF support unless all the critera are met.
+#
+# TODO(camh): Remove /usr/libbpf-1.2.2 when old buildboxes are replaced by ng
+
+with_bpf := no
+BPF_MESSAGE := without-BPF-support
+
+CLANG ?= $(shell which clang || which clang-12)
+LLVM_STRIP ?= $(shell which llvm-strip || which llvm-strip-12)
+
 # libbpf version required by the build.
 LIBBPF_VER := 1.2.2
 
+FOUND_LIBBPF :=
+
+ifneq (,$(wildcard /usr/libbpf-$(LIBBPF_VER)))
+FOUND_LIBBPF := true
+LIBBPF_INCLUDES := /usr/libbpf-$(LIBBPF_VER)/include
+LIBBPF_LIBS := -L/usr/libbpf-$(LIBBPF_VER)/lib64 -lbpf
+# libbpf needs libelf. Try to find it with pkg-config/pkgconf and fallback to
+# hard-coded defaults if pkg-config says nothing.
+LIBBPF_LIBS += $(or $(shell $(PKGCONF) --silence-errors --static --libs libelf),-lelf -lz)
+else
+ifneq (,$(shell $(PKGCONF) --exists 'libbpf = $(LIBBPF_VER)' && echo true))
+FOUND_LIBBPF := true
+LIBBPF_INCLUDES := $(shell $(PKGCONF) --cflags libbpf)
+LIBBPF_LIBS := $(shell $(PKGCONF) --libs --static libbpf)
```

Comment:
This will break our OSS build for some users that have libbpf installed in the system (unless now we only provide the official builds).
The problem with our Go libbpf wrapper is that not backward compatible, so you MUST have the expected version. `pkg-conf` returns flags if any version is installed. 
Normally I'd use pkg-conf for everything, but in this case I don't think we should be using it. There is very small change that your OS have the correct version installed and flags returned by pkg-conf are always related to the system wide installed version.

Reply:
I thought I addressed this a couple of lines above with:
```
ifneq (,$(shell $(PKGCONF) --exists 'libbpf = $(LIBBPF_VER)' && echo true))
```

This block is only executed if the specific version of libbpf is installed with pkgconf. Do you think there can be a problem if multiple versions of libbpf are installed? The man page for pkg-config suggests that using something like `--exact-version=X.Y.Z` can only be used with `--exists`, and even then its unnecessary due to `--exists` supporting version constraints as I've used. Perhaps I can use `--lib` with `--exact-version` to ensure we get the correct libs - I'll test that out.

Reply:
Aaaa, sorry. I missed this part. TBH I don't have strong opinion about it. Feel free to merge it.